### PR TITLE
feat(boojum): Adding `zk status prover` command

### PR DESCRIPTION
**Pull request by mm-zk**
# What ❔

* Adding zk status prover command to see the current state of FRI prover
* mostly focused on local deployments (might require some changes before we run it on boojnet)

## Why ❔

* to better see what's going on in the prover pipeline

## Checklist

```shell
zk status prover
```

image

### CODE REVIEW

File: infrastructure/zk/src/status.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
+import { Command } from 'commander';
+
+import { Pool } from 'pg';
+import { ethers } from 'ethers';
+import { assert } from 'console';
+
+// Postgress connection pool - must be intialized later - as the ENV variables are set later.
+let pool: any = null;
```
Comment (ly0va) [inline infrastructure/zk/src/status.ts:8]: To still get type hints, you can do something like
```suggestion
let pool: Pool | null = null;
```

File: infrastructure/zk/src/status.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+const GETTER_ABI = [
```
Comment (ly0va) [inline infrastructure/zk/src/status.ts:10]: More of a nit/suggestion, but if you want to hardcode the ABIs, ethers supports the following format

```
const GETTER_ABI = [
    'function getTotalBlocksCommitted() returns (uint256)', 
    'function getTotalBlocksVerified() returns (uint256)'
]
```

**Review by ly0va — APPROVED** [review]

File: infrastructure/zk/src/status.ts (hunk 2 above)
Comment (mm-zk) [reply infrastructure/zk/src/status.ts:10]: oh - that's super nice. Thanks !

File: infrastructure/zk/src/status.ts (hunk 1 above)
Comment (mm-zk) [reply infrastructure/zk/src/status.ts]: done.

**Pull request merged**